builtin/logical/ssh/path_issue_sign.go

```diff
@@ -101,15 +98,18 @@ func (b *backend) pathSignIssueCertificateHelper(ctx context.Context, req *logic
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
-	privateKeyEntry, err := caKey(ctx, req.Storage, caPrivateKey)
+	issuerId, err := sc.resolveIssuerReference(role.Issuer)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read CA private key: %w", err)
+		return handleStorageContextErr(err)
 	}
-	if privateKeyEntry == nil || privateKeyEntry.Key == "" {
-		return nil, errors.New("failed to read CA private key")
+	issuer, err := sc.fetchIssuerById(issuerId)
+	if err != nil {
+		return handleStorageContextErr(err)
 	}
 
-	signer, err := ssh.ParsePrivateKey([]byte(privateKeyEntry.Key))
+	privateKey := issuer.PrivateKey
```

Comment:
This will be fun to deal with when it comes time to do PKCS#11 HSM support from the working group, but that's not a you problem now :-)

Reply:
Interesting, how so?

Reply:
We'll need a `crypto.Signer` interface, rather than the actual byte material of the private key; it tends not to be extractable.